WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/domain/GetStats.kt

```diff
@@ -27,14 +30,24 @@ class GetStats @Inject constructor(
     private val localeProvider: LocaleProvider,
     private val statsRepository: StatsRepository,
     private val appPrefsWrapper: AppPrefsWrapper,
-    private val coroutineDispatchers: CoroutineDispatchers
+    private val coroutineDispatchers: CoroutineDispatchers,
+    private val analyticsUpdateDataStore: AnalyticsUpdateDataStore
 ) {
-    suspend operator fun invoke(refresh: Boolean, granularity: StatsGranularity): Flow<LoadStatsResult> =
-        merge(
+    suspend operator fun invoke(refresh: Boolean, granularity: StatsGranularity): Flow<LoadStatsResult> {
+        val selectionType = StatsTimeRangeSelection.SelectionType.from(granularity)
+        val isForcedRefresh = shouldUpdateStats(selectionType, refresh)
+        return merge(
             hasOrders(),
-            revenueStats(refresh, granularity),
-            visitorStats(refresh, granularity)
-        ).flowOn(coroutineDispatchers.computation)
+            revenueStats(isForcedRefresh, granularity),
+            visitorStats(isForcedRefresh, granularity)
+        ).map { result ->
+            if (result is LoadStatsResult.RevenueStatsSuccess && isForcedRefresh) {
+                analyticsUpdateDataStore.storeLastAnalyticsUpdate(selectionType)
+            }
+            result
+        }
+            .flowOn(coroutineDispatchers.computation)
```

Comment:
NP: I think the `onEach` operation would be a better fit here since there's no transformation of the data itself, this way you will not need to return the `result` at the end of the lambda.

```suggestion
        ).onEach { result ->
            if (result is LoadStatsResult.RevenueStatsSuccess && isForcedRefresh) {
                analyticsUpdateDataStore.storeLastAnalyticsUpdate(selectionRange)
            }
        }.flowOn(coroutineDispatchers.computation)
```